src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,331 @@
+              {t('swapReviewScreen.transactionDetails')}
+            </Text>
+            <View style={styles.row}>
+              <View style={styles.column}>
+                <Text style={styles.label}>{t('exchangeRate')}</Text>
+              </View>
+              <View style={styles.column}>
+                <Text style={styles.transactionDetailsRightText}>
+                  {`1 ${fromTokenSymbol} ≈ ${formatValueToDisplay(
+                    new BigNumber(swapInfo?.unvalidatedSwapTransaction?.price)
+                  )} ${toTokenSymbol}`}
+                </Text>
+              </View>
+            </View>
+            <View style={styles.row}>
+              <View style={styles.column}>
+                <Text style={styles.label}>{t('swapReviewScreen.estimatedGas')}</Text>
+              </View>
+              <View style={[styles.column, styles.tokenDisplayView]}>
+                <TokenDisplay
+                  style={styles.transactionDetailsRightText}
+                  amount={divideByWei(swapInfo?.unvalidatedSwapTransaction?.gas)}
+                  tokenAddress={fromToken}
+                  showLocalAmount={false}
+                  testID={'EstimatedGas'}
+                />
+              </View>
+            </View>
+          </View>
+        </ScrollView>
+      )}
+      <Button
+        style={{ padding: Spacing.Regular16 }}
+        onPress={() => Logger.debug('TODO Perform Swap!!!')}
+        text={t('swapReviewScreen.complete')}
+        size={BtnSizes.FULL}
+        disabled={loading}
+      />
+      {/** Estimated amount dialog */}
+      <Dialog
+        title={t('swapReviewScreen.estimatedAmountTitle')}
+        isVisible={estimatedModalVisible}
+        actionText={t('dismiss')}
+        actionPress={() => setEstimatedDialogVisible(false)}
+        isActionHighlighted={false}
+        onBackgroundPress={() => setEstimatedDialogVisible(false)}
+      >
+        {t('swapReviewScreen.estimatedAmountBody', {
+          slippagePercent: maxSlippagePercent,
+        })}
+      </Dialog>
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  safeAreaContainer: {
+    flex: 1,
+  },
+  loadingContentContainer: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  contentContainer: {
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  subContentContainer: {
+    paddingBottom: Spacing.Regular16,
+  },
+  tallRow: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Regular16,
+  },
+  row: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Smallest8,
+  },
+  // Used for large text display
+  column: {
+    maxWidth: '50%',
+  },
+  tokenDisplayView: {
+    justifyContent: 'space-between',
+  },
+  separator: {
+    height: 1,
+    width: '100%',
+    backgroundColor: colors.gray2,
+  },
+  // No shared font styles for this label - done manually
```

Comment:
can we use some base style from the font styles (just the closest one) and update the fields we need? at least this way we can see what is pretty much supposed to look like which styles are overwritten, rather than a whole new font style. (i'd also push back on adding new styles like this, since we have a design system which would have no purpose if we create custom styles whenever we want)

Reply:
Styles adjusted in https://github.com/valora-inc/wallet/pull/2767/commits/70960e345f0debe2cbefb3fce5e70ef19e3fddca!